test/blackbox-tests/test-cases/copy_files/test6.t

```diff
@@ -0,0 +1,27 @@
+Show that copy_files operates on the build folder, copying over e.g. .re.ml files
+
+  $ mkdir -p subdir
+  $ cat >dune-project <<EOF
+  > (lang dune 3.13)
+  > EOF
+  $ cat >subdir/dune <<EOF
+  > (library (name foo))
```

Comment:
Could you change the test not to rely on libraries? It should be easy enough to just use the files being copied as targets.

Reply:
Sorry, i don't understand what you mean. The rule that has `foo.re.ml` as target is created by the library, afaiu. If I remove the library and try to do `dune build subdir/foo.re.ml` (which is what I assume the suggestion is about), dune will show `Error: Don't know how to build subdir/foo.re.ml`.

If I target the copied file `foo.re` directly, then I can't test the difference between regular `copy_files` behavior and the new behavior, because in both cases the target will be generated, because it's in the source.

Could you share a snippet of what you have in mind please?

Reply:
Something like this:
```
$ cat >target/dune <EOF
> (copy_files
>  (only_sources true)
> (files foo/*))
> EOF
 
$ cat >foo/dune <<EOF
> (with-stdout-to in-build (echo ""))
> EOF

$ touch foo/in-source

$ dune build target/in-source # should pass
$ dune build target/in-build # should fail with unknown target
```

Reply:
I tried this but get an error due to multiple rules generated:

```
  $ mkdir -p target foo
  $ cat >dune-project <<EOF
  > (lang dune 3.14)
  > EOF
  $ cat >target/dune <<EOF
  > (copy_files
  >  (files ../foo/*))
  > EOF
  $ cat >foo/dune <<EOF
  > (rule
  >  (with-stdout-to in-build
  >  (echo "")))
  > EOF

  $ touch foo/in-source

  $ dune build target/in-source # should pass
  Error: Multiple rules generated for _build/default/target/dune:
  - target/dune:2
  - file present in source tree
  Hint: rm -f target/dune
  [1]
  $ dune build target/in-build # should fail with unknown target
  Error: Multiple rules generated for _build/default/target/dune:
  - target/dune:2
  - file present in source tree
  Hint: rm -f target/dune
  [1]

Show the difference when `only_sources` is used

  $ cat >target/dune <<EOF
  > (copy_files
  >  (only_sources true)
  >  (files ../foo/*))
  > EOF

  $ dune build target/in-source # should pass
  Error: Multiple rules generated for _build/default/target/dune:
  - target/dune:3
  - file present in source tree
  Hint: rm -f target/dune
  [1]
  $ dune build target/in-build # should fail with unknown target
  Error: Multiple rules generated for _build/default/target/dune:
  - target/dune:3
  - file present in source tree
  Hint: rm -f target/dune
  [1]
```

Reply:
I think you can just modify the glob to exclude `dune`. One simple way is to use an extension for your files and then select them with `*.txt`.